src/pudl/io_managers.py

```diff
@@ -276,6 +276,13 @@ def _handle_pandas_output(self, context: OutputContext, df: pd.DataFrame):
         table_name = self._get_table_name(context)
 
         sa_table = self._get_sqlalchemy_table(table_name)
+        pkg = Package.from_resource_ids()
+        all_resources = [resource.name for resource in pkg.resources]
+        if table_name in all_resources:
+            res = pkg.get_resource(table_name)
+            df = res.enforce_schema(df)
+        else:
+            df = pudl.metadata.fields.apply_pudl_dtypes(df)
 
         column_difference = set(sa_table.columns.keys()) - set(df.columns)
```

Comment:
This line contains duplicates some logic in enforce_schema.

Reply:
Which line, the `column_difference`?  Should that be inside the `else:` then?

Reply:
Okay, now it seems like this should be here, since this IO Manager doesn't use enforce schema.